sources/packages/web/src/components/generic/CheckAESTPermissionRole.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <slot :isReadonly="isReadonly"></slot>
+</template>
+
+<script lang="ts">
+import { computed, PropType } from "vue";
+import { ClientIdType, Role } from "@/types";
+import { AuthService } from "@/services/AuthService";
+
+export default {
+  props: {
+    role: {
+      type: String as PropType<Role>,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const isReadonly = computed(() => {
+      const userToken = AuthService.shared.userToken;
+      if (userToken?.resource_access && userToken?.azp) {
```

Comment:
This logic can be moved to a composable that, based on the current "azp" will just return the list of the roles.
I do not see a reason for this being AEST specific, but maybe I am missing something.